openquake/hazardlib/near_fault.py

```diff
@@ -0,0 +1,476 @@
+        A positive angle is generated by a counter-clockwise rotation.
+    :return:
+        fs, float value of the average S-wave radiation pattern.
+        rd, the distance from site to the direct point.
+        r_hyp, the hypocetre distance. Both distance calculation has been
+        sugggested in the Spudich, et al.(2013)
+    """
+    # zs
+
+    site_xyz = get_xyz_from_ll(site, reference)
+    zs = dst.pdist([pp, site_xyz])
+
+    if site_xyz[0] * normal[0] + site_xyz[1] * normal[1] + site_xyz[2] * \
+       normal[2] - dist_to_plane > 0:
+        zs = -zs
+
+    # l2
+
+    hyp_xyz = get_xyz_from_ll(hypocenter, reference)
+    hyp_xyz = np.array(hyp_xyz).reshape(1, 3).flatten()
+    l2 = dst.pdist([pp, hyp_xyz])
+
+    # Rd
+
+    rd = ((l2 - e) ** 2 + zs ** 2) ** 0.5
+
+    # Rhyp
+
+    r_hyp = (l2 ** 2 + zs ** 2) ** 0.5
+
+    # phi
+
+    p0_xyz = get_xyz_from_ll(p0, reference)
+    p1_xyz = get_xyz_from_ll(p1, reference)
+    u = (np.array(p1_xyz) - np.array(p0_xyz))
+    v = pp - hyp_xyz
+    phi = vectors2angle(u, v) - np.deg2rad(delta_slip)
+
+    # Ix
+
+    ix = np.cos(phi) * (2 * zs * (l2 / r_hyp - (l2 - e) / rd) -
+                        zs * np.log((l2 + r_hyp) / (l2 - e + rd)))
+
+    # In
+
+    inn = np.cos(phi) * (-2 * zs ** 2 * (1 / r_hyp - 1 / rd)
+                         - (r_hyp - rd))
+
+    # Iphi
+
+    iphi = np.sin(phi) * (zs * np.log((l2 + r_hyp) / (l2 - e + rd)))
+
+    # FS
+
+    fs = (ix ** 2 + inn ** 2 + iphi ** 2) ** 0.5 / e
+
+    return fs, rd, r_hyp
+
+
+def isochone_ratio(e, rd, r_hyp):
+    """
+    Get the isochone ratio which desrcibed in: "Final
+    report of the NGA-West2 directivity working group." (2013) by Spudich,
+    Paul, et al. and publish in PEER report, page 88.
+    :param e:
+        E-path, in km.
+    :param rd:
+        float, distance from the site to the direct point.
+    :param r_hyp:
+        float, the hypocentre distance.
+    :return:
+        c_prime, float valut, the isochone ration.
+    """
+
+    if e == 0.:
+        c_prime = 0.8
+    elif e > 0.:
+        c_prime = 1. / ((1. / 0.8) - ((r_hyp - rd) / e))
+
+    return c_prime
+
+
+def _intersection(seg1_start, seg1_end, seg2_start, seg2_end):
+    """
+    Get the intersection point in the space when given the 2 line segments.
+    :param seg1_start:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing one end point of a segment(e.g. segment1)
+        segment.
+    :param seg1_end:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the other end point of the first segment(e.g. segment1)
+    :param seg2_start:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing one end point of the other segment(e.g. segment2)
+        segment.
+    :param seg2_end:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the other end point of the second segment(e.g. segment2)
+    :return:
```

Comment:
changed.
I want to say it is a numpy array.
Now, it changes to 
vector1, a numpy array, .....